bluemira/builders/thermal_shield.py

```diff
@@ -0,0 +1,356 @@
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Thermal shield builders
+"""
+
+from typing import List, Tuple, Type
+
+import numpy as np
+from scipy.spatial import ConvexHull
+
+from bluemira.base.builder import Builder, ComponentManager
+from bluemira.base.components import PhysicalComponent
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import NewParameter as Parameter
+from bluemira.base.parameter_frame import parameter_frame
+from bluemira.builders.tools import (
+    circular_pattern_component,
+    find_xy_plane_radii,
+    get_n_sectors,
+    make_circular_xy_ring,
+)
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.error import GeometryError
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.plane import BluemiraPlane
+from bluemira.geometry.tools import (
+    boolean_cut,
+    boolean_fuse,
+    make_polygon,
+    offset_wire,
+    revolve_shape,
+)
+from bluemira.geometry.wire import BluemiraWire
+
+
+class VacuumVesselThermalShield(ComponentManager):
+    """
+    Wrapper around a VacuumVesselThermalShield component tree.
+    """
+
+
+@parameter_frame
+class VVTSBuilderParams:
+    """
+    VVTS builder parameters
+    """
+
+    g_vv_ts: Parameter[float]
+    n_TF: Parameter[int]
+    tk_ts: Parameter[float]
+
+
+class VVTSDesigner(Designer[BluemiraWire]):
+    """
+    Vacuum vessel thermal shield designer
+    """
+
+    param_cls = None
+
+    def __init__(
+        self,
+        vv_koz: BluemiraWire,
+    ):
+        super().__init__()
+        self.vv_koz = vv_koz
+
+    def run(self):
+        """
+        Vacuum vessel thermal shield designer run method
+        """
+        return self.vv_koz
+
+
+class VVTSBuilder(Builder):
+    """
+    Vacuum vessel thermal shield builder
+    """
+
+    VVTS = "VVTS"
+    param_cls: Type[VVTSBuilderParams] = VVTSBuilderParams
+
+    def build(self) -> VacuumVesselThermalShield:
+        """
+        Build the vacuum vessel thermal shield component.
+        """
+        koz = self.designer.run()
```

Comment:
I've been adding an `__init__` in all concrete builders `Builder`s, partially for typing the designer, but mainly because the `designer` argument in the base class is optional, but in the concrete classes it's either needed or it's not.

Reply:
sure, done